src/mmw/apps/modeling/views.py

```diff
@@ -373,7 +382,11 @@ def get_list_of_workers():
 
     workers = filter(predicate,
                      get_list_of_workers())
-    return random.choice(workers)
+    return workers
+
+
+def choose_worker():
```

Comment:
One potential downside to changing the way this works is that the list of workers `choose_worker()` operates on grows stale as work is being done. I don't know that that is super likely, but are we currently asking Celery for available workers enough that this optimization gets us a lot?